### PR TITLE
get dns config for esxi hosts

### PR DESCRIPTION
**Pull request by cheburakshu**
#ssc-119

### CODE REVIEW

File: src/saltext/vmware/modules/esxi.py [hunk 1]
```diff
@@ -732,3 +732,62 @@ def set_advanced_config(
...
+    log.debug("Running vmware_esxi.get_dns_config")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=True if not host_name else False,
```
Comment (waynew) [inline src/saltext/vmware/modules/esxi.py:771]: **question** would `bool(host_name)` read better here?

File: src/saltext/vmware/modules/esxi.py [hunk 2]
```diff
@@ -732,3 +732,62 @@ def set_advanced_config(
...
+    )
+
+    try:
+        for h in hosts:
+            dns_config = h.config.network.dnsConfig
+            ret[h.name] = {}
```
Comment (waynew) [inline src/saltext/vmware/modules/esxi.py:777]: **question** is it intentional that there will be an empty dict even when `not dns_config`? in other words, should this come after the `if not dns_config` block?

File: tests/integration/modules/test_esxi.py [hunk 3]
```diff
@@ -264,3 +264,18 @@ def test_advanced_config(service_instance):
...
+def test_get_dns_config(service_instance):
+    """
+    Test get dns configuration on ESXi host
+    """
+    ret = esxi.get_dns_config(
+        service_instance=service_instance,
+        datacenter_name="Datacenter",
+        cluster_name="Cluster",
+    )
+    for host in ret:
```
Comment (waynew) [inline tests/integration/modules/test_esxi.py:278]: **suggestion** note that this will pass without doing anything if there is nothing in `ret`, which would happen if `utils_esxi.get_hosts` returns empty. We should have either a unit test to confirm the behavior always (for instance, mocking get_hosts to return something expected/unexpected and ensure that the expected results are returned or the expected exception handling happens), or this test should probably fail when there are no results (since testing nothing is a useless test)

**Review by waynew — CHANGES_REQUESTED** [review]
Mostly looks good - couple of minor items

File: src/saltext/vmware/modules/esxi.py (hunk 1 above)
Comment (cheburakshu) [reply src/saltext/vmware/modules/esxi.py]: done

File: tests/integration/modules/test_esxi.py (hunk 3 above)
Comment (cheburakshu) [reply tests/integration/modules/test_esxi.py:278]: done

Comment (waynew) [reply tests/integration/modules/test_esxi.py:278]: :thinking: I'm not sure if I'm following - I see that the other test is checking when `ret` is *supposed* to be empty, but what about the first case when we expect it to be something?

Would 

```
assert ret
for host in ret:
 ...
 ```
 
 Be appropriate/correct?

**Comment by cheburakshu** [comment]
> Other changes look good, just not sure about the test change

I added one more test to handle the previous comment when the function returns an empty dict when no host is matched. When one or more hosts are matched, it will return a dictionary of dns info indexed by host name.

> note that this will pass without doing anything if there is nothing in ret
